lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -493,3 +503,66 @@ txOutMinTokenQuantity = TokenQuantity 1
 --
 txOutMaxTokenQuantity :: TokenQuantity
 txOutMaxTokenQuantity = TokenQuantity $ fromIntegral $ maxBound @Word64
+
+--------------------------------------------------------------------------------
+-- Constraints
+--------------------------------------------------------------------------------
+
+-- | Provides an abstract cost model for transactions.
+--
+-- This allows parts of a transaction to be costed individually, without
+-- having to compute the cost of an entire transaction.
+--
+data TxConstraints s = TxConstraints
+    { txBaseCost :: Coin
+      -- ^ The constant cost of an empty transaction.
+    , txBaseSize :: s
```

Comment:
I agree this would reduce the number of fields, but I'm not entirely convinced the benefit would exceed the cost!

In particular, I find the revised field names to be less clear. For example, at first glance, `txInput` seems like it ought to return an input, but actually it's the cost and size **_of_** an input. IMO, `txInputSize` and `txInputCost` seem more straightforward and easy to understand.